os/arch/arm/src/amebasmart/amebasmart_i2c.c

```diff
@@ -1163,6 +1163,10 @@ FAR struct i2c_dev_s *up_i2cinitialize(int port)
 		return NULL;
 	}
 
+	/* If I2C has been initialized, return directly */
```

Comment:
Hi @kishore-sn , if there is multiple attempt for initialisation, the condition here will return directly.